internal/wasm/func_validation.go

```diff
@@ -868,9 +868,19 @@ func (m *Module) validateFunctionWithMaxStackValues(
 			pc += num - 1
 		} else if op == OpcodeMiscPrefix {
 			pc++
-			// Miscellaneous instructions come with two bytes which starts with OpcodeMiscPrefix,
-			// and the second byte determines the actual instruction.
-			miscOpcode := body[pc]
+			// A misc opcode is encoded as an unsigned variable 32-bit integer.
+			miscOp32, num, err := leb128.LoadUint32(body[pc:])
```

Comment:
I didn't know that amazingly misc opcodes are encoded in leb128 U32...

Reply:
oh good that you found this, though scary to expect to need this encoding :D